webpack.config.js

```diff
@@ -145,34 +156,44 @@ module.exports = ( env ) => {
       }),
 
       // Uglify bundles
-      ifProd( new webpack.optimize.UglifyJsPlugin( {
+      ifProdOrSite(new webpack.optimize.UglifyJsPlugin( {
         compress: {
           screw_ie8: true,
           warnings: false
         },
         output: { comments: false }
       } ) ),
 
-      ifProd( new FaviconsWebpackPlugin( './assets/blaze-elements-logo.svg' ) ),
+      ifDevOrSite( new FaviconsWebpackPlugin( './assets/blaze-elements-logo.svg' ) ),
 
       /**
        * Use the HtmlWebpackPlugin plugin to make index.html a template so css and js can dynamically be added to the page.
        * This will also take care of moving the index.html file to the build directory using the index.html in src as a template.
        * https://github.com/ampedandwired/html-webpack-plugin
        */
-      ifNotTest(new HtmlWebpackPlugin({
+      ifDevOrSite(new HtmlWebpackPlugin({
           template: resolve( 'index.html' ),
           packages: env.element ? env.element : require('./package.json').packages,
-          excludeChunks: [ 'main' ], // Exclude 'main' as it is included in 'main.demo'
+          excludeChunks: [ 'index', 'index-with-dependencies' ], // Exclude 'index' & 'index-with-dependencies' as it is included in 'main.demo'
           inject: 'head',
-          chunksSortMode: buildChunksSort([ 'polyfills', 'vendors', 'styles', 'main', 'main.demo', 'test-helpers', 'test' ])
+          chunksSortMode: buildChunksSort([ 'polyfills', 'styles', 'index', 'index-with-dependencies', 'main.demo', 'test-helpers', 'test' ])
       }))
 
     ]),
     performance: {
       hints: ifTest() ? false : 'warning'
     }
   }
+
+
+  function ifProdOrSite( callback ) {
+    return ifProd( callback, ifSite( callback ) );
```

Comment:
wow looks like you have tried to open gate to callback hell dimension 😄 

![mr-clippy](https://cloud.githubusercontent.com/assets/82104/22650516/d4414258-ec7f-11e6-9f36-8a95b0e6a38e.jpg)